src/MessageCreationUtil.js

```diff
@@ -22,6 +30,23 @@ export default class MessageCreationUtil {
         return this.usernamePromise
     }
 
+    async getStream(streamId) {
+        if (!this.cachedStreams.get(streamId)) {
+            if (!this.ongoingRequests[streamId]) {
+                this.ongoingRequests[streamId] = this.getStreamFunction(streamId)
+            }
+            const stream = await this.ongoingRequests[streamId]
+            const success = this.cachedStreams.set(streamId, {
+                id: stream.id,
+                partitions: stream.partitions,
+            })
```

Comment:
So `cachedStreams` is supposed to evict the stream after 30 minutes, presumably to prevent memory leaks, but the stream can never be garbage collected because a reference will still be held in the `this.ongoingRequests[streamId]` promise. 

As it stands, you could actually just remove `node-cache` because `ongoingRequests` is already acting like a cache for you, only thing it's not doing is eviction.

Anyway, it should probably do something like:
```js
let stream
try {
   stream = await this.ongoingRequests[streamId]
} finally {
  delete this.ongoingRequests[streamId]
}
```